src/v/config/node_config.cc

```diff
@@ -191,6 +191,25 @@ node_config::node_config() noexcept
                                  "and write permissions"};
           }
 
+          return std::nullopt;
+      })
+  , rpk_path(
+      *this,
+      "rpk_path",
+      "The path to the RPK binary. Used to create debug bundles (default: "
+      "/usr/bin/rpk)",
+      {.visibility = visibility::user},
+      "/usr/bin/rpk",
+      [](const std::filesystem::path& bin_path) -> std::optional<ss::sstring> {
+          if (!std::filesystem::exists(bin_path)) {
+              return ss::sstring{"RPK path does not exist"};
+          }
+
+          // access is apart of unistd.h
+          if (access(bin_path.c_str(), X_OK) != 0) {
+              return ss::sstring{"RPK path does not have exec permissions"};
+          }
```

Comment:
should we split this validation out to run somewhere else like application.cc? i would assume that this isn't a single path, but a set of paths, including looking it up in `PATH` env?

Reply:
This is just looking for the `rpk` binary to run.  I don't believe `posix_spawn` will honor the process's PATH environment to look for `rpk`.

I suggested adding this configuration for cases (like during CI) when `rpk` isn't located in `/usr/bin` or in `PATH`

Reply:
> should we split this validation out to run somewhere else like application.cc?

I do not understand the risks or design flaws of running this validation in the config store.

> i would assume that this isn't a single path, but a set of paths, including looking it up in PATH env?

I did not consider looking up the rpk path in the PATH env.

> This is just looking for the rpk binary to run. I don't believe posix_spawn will honor the process's PATH environment to look for rpk.

Correct on both counts. In CI, rpk is not in the PATH so I needed a way to capture that path in order for `ss::process::spawn` to run the binary.